sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -110,25 +110,30 @@ public SentryTracer(
     this.transactionFinishedCallback = transactionFinishedCallback;
 
     if (idleTimeout != null) {
-      scheduleFinish(idleTimeout);
+      timer = new Timer(true);
+      scheduleFinish();
+    } else {
+      timer = null;
     }
   }
 
   @Override
-  public void scheduleFinish(final @NotNull Long idleTimeout) {
+  public void scheduleFinish() {
     cancelTimer();
-    isFinishTimerRunning.set(true);
-    timerTask =
-        new TimerTask() {
-          @Override
-          public void run() {
-            final SpanStatus status = getStatus();
-            finish((status != null) ? status : SpanStatus.OK);
-            isFinishTimerRunning.set(false);
-          }
-        };
-
-    timer.schedule(timerTask, idleTimeout);
+    if (timer != null) {
+      isFinishTimerRunning.set(true);
+      timerTask =
+          new TimerTask() {
```

Comment:
On Android we have a tracer bound to the scope which can be accessed concurrently. We need to synchronize this